### PR TITLE
step functions API

### PR DESCRIPTION
**Pull request by tumdash**
SF API
Test

### CODE REVIEW

File: src/erlcloud_sf.erl [hunk 1]
```diff
@@ -0,0 +1,630 @@
...
+         update_state_machine/3, update_state_machine/1]).
+
+%%------------------------------------------------------------------------------
+%% Library initialization.
+%%------------------------------------------------------------------------------
+-spec new(string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                retry             = fun erlcloud_retry:default_retry/1}.
```
Comment (motobob) [inline src/erlcloud_sf.erl]: nope. please keep one default which is no retry

File: include/erlcloud_aws.hrl [hunk 2]
```diff
@@ -55,6 +55,9 @@
           lambda_scheme="https://"::string(),
           lambda_host="lambda.us-east-1.amazonaws.com"::string(),
           lambda_port=443::non_neg_integer(),
+          sf_scheme="https://"::string(),
+          sf_host="states.us-east-1.amazonaws.com"::string(),
```
Comment (motobob) [inline include/erlcloud_aws.hrl]: can we call all this stepfun_host or states_host...
no value to safe on chars.

File: src/erlcloud_sf.erl [hunk 3]
```diff
@@ -0,0 +1,630 @@
+-module(erlcloud_sf).
```
Comment (motobob) [inline src/erlcloud_sf.erl]: can we please call module erlcloud_stepfun or erlcloud_states

File: src/erlcloud_sf.erl [hunk 4]
```diff
@@ -0,0 +1,630 @@
...
+
+-spec create_state_machine(Definition       :: map(),
+                           StateMachineName :: binary(),
+                           RoleArn          :: binary(),
+                           Config           :: aws_config()) ->
+    {ok, map()} | {error, any()}.
+create_state_machine(Definition, StateMachineName, RoleArn, Config)
+        when is_binary(StateMachineName), is_binary(RoleArn), is_map(Definition) ->
+    Req = #{
+        <<"definition">>    => jsx:encode(Definition),
```
Comment (motobob) [inline src/erlcloud_sf.erl]: should encode nested maps instead of letting jsx do it later?

File: src/erlcloud_sf.erl [hunk 5]
```diff
@@ -0,0 +1,630 @@
...
+%%
+%%------------------------------------------------------------------------------
+-spec list_executions(StateMachineArn :: binary()) -> {ok, map()} | {error, any()}.
+list_executions(StateMachineArn) ->
+    list_executions(StateMachineArn, #{}, default_config()).
+
+-spec list_executions(StateMachineArn   :: binary(),
+                      Options           :: map(),
+                      Config            :: aws_config()) -> {ok, map()} | {error, any()}.
+list_executions(StateMachineArn, Options, Config)
```
Comment (motobob) [inline src/erlcloud_sf.erl]: should there be arrity /2 functions?

File: test/erlcloud_sf_tests.erl [hunk 6]
```diff
@@ -0,0 +1,302 @@
+-module(erlcloud_sf_tests).
```
Comment (motobob) [inline test/erlcloud_sf_tests.erl]: rename in a similar fashion

File: src/erlcloud_sf.erl (hunk 4 above)
Comment (tumdash) [reply src/erlcloud_sf.erl]: in fact that should be JSON-in-JSON, so we need to do it twice

**Pull request merged**